pilot/pkg/security/authz/policy/helper.go

```diff
@@ -125,7 +125,11 @@ func SimpleRole(name string, namespace string, service string) *model.Config {
 }
 
 func BindingTag(name string) string {
-	return fmt.Sprintf("UserFromBinding[%s]", name)
+	return fmt.Sprintf("cluster.local/ns/%s/sa/%s", name, name)
+}
+
+func BindingPrincipal(trustDomain, namespace, saName string) string {
+	return fmt.Sprintf("%s/ns/%s/sa/%s", trustDomain, namespace, saName)
```

Comment:
the namespace doesn't matter here so I would suggest to only pass trustDomain and sa to simplify the name.